app/src/main/java/by/paranoidandroid/dailyvisualizer/view/fragments/SearchFragment.java

```diff
@@ -18,6 +32,43 @@ public void onCreate(Bundle savedInstanceState) {
 
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
-        return inflater.inflate(R.layout.fragment_search, container, false);
+        View view =  inflater.inflate(R.layout.fragment_search, container, false);
+        model = ViewModelProviders.of(getActivity()).get(SearchViewModel.class);
+
+        tvNothingToShow = view.findViewById(R.id.tv_nothing_to_show);
+        recyclerView = view.findViewById(R.id.rv_search);
+        timeLineDayRecyclerAdapter = new TimeLineDayRecyclerAdapter();
+        recyclerView.setAdapter(timeLineDayRecyclerAdapter);
+        progressBar = view.findViewById(R.id.pb_search);
+
+        searchView = view.findViewById(R.id.search_view);
+        searchView.setOnQueryTextListener(new OnQueryTextListener() {
+            @Override
+            public boolean onQueryTextSubmit(String s) {
+                progressBar.setVisibility(View.VISIBLE);
+                recyclerView.setVisibility(View.GONE);
+                tvNothingToShow.setVisibility(View.GONE);
+                model.setSearchQuery(s);
```

Comment:
I added a new commit to this branch:
"A little fix: trim() method added (to remove any leading and trailing whitespaces)"
https://github.com/Kalyanova/DailyVisualizer/pull/18/commits/e6cd269fe62e8c607ddc4e5274b2bcd02d641842

Reply:
And one more commit to this branch :) To clear SearchView by clicking on close button.
"Added OnCloseListener on SearchView"
https://github.com/Kalyanova/DailyVisualizer/pull/18/commits/db454bafc95b39983951ad24d8c77f98667f8102